core/src/library.rs

```diff
@@ -193,6 +199,35 @@ impl<'gc> MovieLibrary<'gc> {
     pub fn set_device_font(&mut self, font: Option<Font<'gc>>) {
         self.device_font = font;
     }
+
+    /// Check if the current movie's VM tendency is compatible with running
+    /// code on a particular VM. If it is not, then this yields an error.
+    ///
+    /// Checking the VM tendency will also set the VM tendency for the entire
+    /// movie if it is not already set. This ensures that, say, a movie can't
+    /// claim it's AS3 in it's file attributes, but then start running AS2
+    /// code.
+    pub fn check_vm_tendency(&mut self, new_tendency: AvmType) -> Result<(), Error> {
+        if self.vm_tendency.map(|t| t != new_tendency).unwrap_or(false) {
+            return Err(format!(
+                "Blocked attempt to run {:?} code on an {:?} movie.",
+                new_tendency,
+                self.vm_tendency.unwrap()
+            )
+            .into());
+        }
+
+        self.vm_tendency = Some(new_tendency);
+
+        Ok(())
```

Comment:
The library should have it's AVM version set based on the FileAttributes tag when we parse it as part of the loading.